Libraries/Animated/__tests__/Animated-web-test.js

```diff
@@ -0,0 +1,187 @@
+      const translateAnim = anim.interpolate({
+        inputRange: [0, 1],
+        outputRange: [100, 200],
+      });
+
+      const callback = jest.fn();
+
+      const node = new AnimatedProps(
+        {
+          style: {
+            backgroundColor: 'red',
+            opacity: anim,
+            transform: [
+              {
+                translate: [translateAnim, translateAnim],
+              },
+              {
+                translateX: translateAnim,
+              },
+              {scale: anim},
+            ],
+            shadowOffset: {
+              width: anim,
+              height: anim,
+            },
+          },
+        },
+        callback,
+      );
+
+      expect(node.__getValue()).toEqual({
+        style: [
+          {
+            backgroundColor: 'red',
+            opacity: anim,
+            shadowOffset: {
+              width: anim,
+              height: anim,
+            },
+            transform: [
+              {translate: [translateAnim, translateAnim]},
+              {translateX: translateAnim},
+              {scale: anim},
+            ],
+          },
+          {
+            opacity: 0,
+            transform: [{translate: [100, 100]}, {translateX: 100}, {scale: 0}],
+            shadowOffset: {
+              width: 0,
+              height: 0,
+            },
+          },
+        ],
+      });
+
+      expect(anim.__getChildren().length).toBe(0);
+
+      node.__attach();
+
+      expect(anim.__getChildren().length).toBe(3);
+
+      anim.setValue(0.5);
+
+      expect(callback).toBeCalled();
+
+      expect(node.__getValue()).toEqual({
+        style: [
+          {
+            backgroundColor: 'red',
+            opacity: anim,
+            shadowOffset: {
+              width: anim,
+              height: anim,
+            },
+            transform: [
+              {translate: [translateAnim, translateAnim]},
+              {translateX: translateAnim},
+              {scale: anim},
+            ],
+          },
+          {
+            opacity: 0.5,
+            transform: [
+              {translate: [150, 150]},
+              {translateX: 150},
+              {scale: 0.5},
+            ],
+            shadowOffset: {
+              width: 0.5,
+              height: 0.5,
+            },
+          },
+        ],
+      });
+
+      node.__detach();
+      expect(anim.__getChildren().length).toBe(0);
+
+      anim.setValue(1);
+      expect(callback.mock.calls.length).toBe(1);
+    });
+
+    it('does not discard initial style', () => {
+      const value1 = new Animated.Value(1);
+      const scale = value1.interpolate({
+        inputRange: [0, 1],
+        outputRange: [1, 2],
+      });
+      const callback = jest.fn();
+      const node = new AnimatedProps(
+        {
+          style: {
+            transform: [
+              {
+                scale,
+              },
+            ],
+          },
+        },
+        callback,
+      );
```

Comment:
This test will be a bit confusing without changes / explanation. The behavior matters when the input style is a mix of values from `StyleSheet.create` (ie contains no animated values, could be optimized to another format) and an inline style with an animation

Reply:
Sounds good, I've updated tests a bit to include `StyleSheet.create`